src/analysis/special_functions/polynomials.lean

```diff
@@ -147,4 +156,21 @@ begin
     exact tendsto_abs_at_bot_at_top.comp (P.div_tendsto_at_bot_of_degree_gt Q hdeg hQ h.le) }
 end
 
+theorem is_O_of_degree_le (h : P.degree ≤ Q.degree) :
+  is_O (λ x, eval x P) (λ x, eval x Q) filter.at_top :=
+begin
+  by_cases hp : P = 0,
+  { simpa [hp] using is_O_zero (λ x, eval x Q) filter.at_top },
+  { have hq : Q ≠ 0 := ne_zero_of_degree_ge_degree h hp,
+    cases le_iff_lt_or_eq.mp h with h h,
+    { have := polynomial.div_tendsto_zero_of_degree_lt P Q h,
+      refine is_O_at_top_of_div_tends_to_finite _ 0 this,
+      refine filter.mem_sets_of_superset (polynomial.eventually_no_roots Q hq) _,
+      exact λ x hx hx', absurd hx' hx },
+    { have := polynomial.div_tendsto_leading_coeff_div_of_degree_eq P Q h,
+      refine is_O_at_top_of_div_tends_to_finite _ _ this,
+      refine filter.mem_sets_of_superset (polynomial.eventually_no_roots Q hq) _,
+      exact λ x hx hx', absurd hx' hx } }
+end
```

Comment:
I think this could be simplified a bit by using `polynomial.is_equivalent_at_top_div`. Then you would only have to prove that `λ (x : 𝕜), (P.leading_coeff / Q.leading_coeff) * x ^ (↑(P.nat_degree) - ↑(Q.nat_degree))` tends to some finite number. which should not be hard using `tendsto_fpow_at_top_zero` and `fpow_zero`.